std/variant.d

```diff
@@ -339,7 +339,17 @@ private:
             static if (target.size < A.sizeof)
             {
                 if (target.type.tsize < A.sizeof)
-                    *cast(A**)&target.store = new A;
+                {
+                    static if (is(A == U[n], U, size_t n))
+                    {
+                        A* p = (new U[n]).ptr;
```

Comment:
I'm sorry, I was completely wrong when I said the `cast(A*)`  could be removed. The problem is that for historical reasons, the expression `new U[n]` produces a `U[]`, not a `U[n]` and so the resulting `ptr` has the type `U*` instead of `U[n]*` which can't be implicitly converted to `U[n]*`, hence the need for a cast. Can you change it back and squash commits?